c7n/resources/rdscluster.py

```diff
@@ -0,0 +1,205 @@
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_snapshot_retention,
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception setting RDS cluster retention  \n %s" % (
+                                f.exception()))
+
+    def process_snapshot_retention(self, resource):
+        current_retention = int(resource.get('BackupRetentionPeriod', 0))
+        new_retention = self.data['days']
+
+        if current_retention < new_retention:
+            self.set_retention_window(
+                resource,
+                max(current_retention, new_retention))
+            return resource
+
+    def set_retention_window(self, resource, retention):
+        c = local_session(self.manager.session_factory).client('rds')
+        c.modify_db_cluster(
+            DBClusterIdentifier=resource['DBClusterIdentifier'],
+            BackupRetentionPeriod=retention,
+            PreferredBackupWindow=resource['PreferredBackupWindow'],
+            PreferredMaintenanceWindow=resource['PreferredMaintenanceWindow'])
+
+
+@actions.register('snapshot')
+class Snapshot(BaseAction):
+
+    schema = type_schema('snapshot')
+
+    def process(self, resources):
+        with self.executor_factory(max_workers=3) as w:
+            futures = []
+            for resource in resources:
+                futures.append(w.submit(
+                    self.process_cluster_snapshot,
+                    resource))
+                for f in as_completed(futures):
+                    if f.exception():
+                        self.log.error(
+                            "Exception creating RDS cluster snapshot  \n %s" % (
+                                f.exception()))
+        return resources
+
+    def process_cluster_snapshot(self, resource):
+        c = local_session(self.manager.session_factory).client('rds')
+        c.create_db_cluster_snapshot(
+            DBClusterSnapshotIdentifier=snapshot_identifier(
+                'Backup',
+                resource['DBClusterIdentifier']),
+            DBClusterIdentifier=resource['DBClusterIdentifier'])
+
+
+@resources.register('rds-cluster-snapshot')
+class RDSClusterSnapshot(QueryResourceManager):
+    """Resource manager for RDS cluster snapshots.
+    """
+
+    class Meta(object):
+
+        service = 'rds'
+        type = 'rds-cluster-snapshot'
+        enum_spec = ('describe_db_cluster_snapshots', 'DBClusterSnapshots', None)
+        name = id = 'DBClusterSnapshotIdentifier'
+        filter_name = None
+        filter_type = None
+        dimension = None
+        date = 'SnapshotCreateTime'
+
+    resource_type = Meta
+
+    filter_registry = FilterRegistry('rdscluster-snapshot.filters')
+    action_registry = ActionRegistry('rdscluster-snapshot.actions')
+
+
+@RDSClusterSnapshot.filter_registry.register('age')
```

Comment:
fwiw, this is fine, but we not have generic age capabilities on the value filter using value_type: age